### ISSUE TITLE
C++ API documentation not updated on branch after a new point release

### ISSUE THREAD

**Issue by Gustry** [qgis/QGIS#43779]
## Description

Looking at the page : 
https://qgis.org/api/3.10/classQgsVectorFileWriter.html#aa69d7084007503269158d425d2709a2e
which is designed for "3.10" in the URL shows only :

> QGIS API Documentation  3.10.0-A Coruña (6c816b4204)

at the top

In the meantime, 3.10.X got new functions.
For instance, `writeAsVectorFormatV2` since 3.10.3

I think documentation on branch should be updated to reflect new/deprecated API, not only .0.

**Comment by roya0045** [comment]
Aren't API changes and new API features mostly relegated to new releases?

**Comment by Gustry** [comment]
Yes but why not regenerating the documentation on a point release ?

I guess, the documentation is triggered automatically on master, so we could add older qgis branch ?

A lot of bugfix commit (changing docstring, new functions, are still backported to older branch, especially after .0).
The situation of `writeAsVectorFormatV2` is a good example. Not possible to find this function on https://qgis.org/api/3.10/classQgsVectorFileWriter.html#aa69d7084007503269158d425d2709a2e because it has been added in 3.10.3
